pkg/kubectl/service_basic.go

```diff
@@ -189,9 +189,6 @@ func (s ServiceCommonGeneratorV1) validate() error {
 	if s.ClusterIP == api.ClusterIPNone && s.Type != api.ServiceTypeClusterIP {
 		return fmt.Errorf("ClusterIP=None can only be used with ClusterIP service type")
 	}
-	if s.ClusterIP == api.ClusterIPNone && len(s.TCP) > 0 {
```

Comment:
Tim, what do you mean by `identity mapped`? Is it that port and targetPort should be equal for headless service, right?